src/kernel/log.h

```diff
@@ -0,0 +1,19 @@
+typedef struct klog_dump {
+    u8 header[4];
+    s64 boot_id;
+    int exit_code;
```

Comment:
This is might be too nitpicky, but for portability on ILP64 platforms, to keep the header size correct it might be better to use a fixed size type for exit_code.

Reply:
Good point, I made `exit_code` an s32